### ISSUE TITLE
Implement message network framework

### ISSUE THREAD

**Issue by NuSkooler** [NuSkooler/enigma-bbs#49]
A message network framework needs implemented. This will allow for various networks (e.g. FidoNet/FTN, QWK based, etc.) to integrate & import/export with the system. I have a design I'm working on for this. This bug is for tracking / relating existing issues.

Some general notes on what's coming:
 * The system will be "layered" allowing different import/export processors (e.g. tossers/etc.) to exist in an abstract way. 
  * Conferences are being introduced. These are simply organizational groups of message areas
  * All `area_name` stuff in DB/code is being renamed to `area_tag`
  * A new configuration structure for conferences, areas, and message network mappings will be introduced (e.g. for `config.hjson`)
  * Initial proof of concept will be FTN processing. Next will very likely be QWK.

Current design for `menu.hjson` message areas & conferences is something like such:
```hjson
conferences: {
	agn: {
		name: Agoranet
		desc: The Official Network of ACiD Productions
		
		//	simple ACS for access
		acs: THEACSSTRING

		areas: {
			agn_bbs: {
				name: BBS Discussion
				desc: General BBS discussion

				//	ACS for what users can do here
				acs: {
					read: THEACSSTRING
					write: THEACSSTRING
					manage: THEACSSTRING
				}
				default: true
			}
		}
	}
}
````

**Comment by NuSkooler** [comment]
For anyone interested, this is being worked on in the [msg_network](https://github.com/NuSkooler/enigma-bbs/tree/msg_network) branch

**Comment by NuSkooler** [comment]
The  [msg_network](https://github.com/NuSkooler/enigma-bbs/tree/msg_network) branch has been merged to master @ 6f8f8f7e9d22c06d04e045ec455c344717d16f9b

There are likely bugs of course. Those will be separate issues.
